File: src/basic/day08/Question01.java
package basic.day08;

public class Question01 {

	public static class Solution {

		public boolean solution(boolean x1, boolean x2, boolean x3, boolean x4) {

			return (x1 || x2) && (x3 || x4);
		}
	}

	public static void main(String[] args) {

		Solution solution = new Solution();

		System.out.println(solution.solution(false, true, true, true));
		System.out.println(solution.solution(true, false, false, false));
	}
}

File: src/beginner/day20/Question01.kt
package beginner.day20

import kotlin.math.abs

class Question01{

    fun solution(dots: Array<IntArray>): Int {

        var maxWidth = Int.MIN_VALUE
        var minWidth = Int.MAX_VALUE
        var maxLength = Int.MIN_VALUE
        var minLength = Int.MAX_VALUE

        for (i in dots) {
            if (maxWidth < i[0]) maxWidth = i[0]
            if (minWidth > i[0]) minWidth = i[0]
            if (maxLength < i[1]) maxLength = i[1]
            if (minLength > i[1]) minLength = i[1]
        }

        return (maxWidth - minWidth) * (maxLength - minLength)
    }
}

fun main() {

    val solution = Question01()

    val dots1 = arrayOf(intArrayOf(1, 1), intArrayOf(2, 1), intArrayOf(2, 2), intArrayOf(1, 2))
    val dots2 = arrayOf(intArrayOf(-1, -1), intArrayOf(1, 1), intArrayOf(1, -1), intArrayOf(-1, 1))

    println(solution.solution(dots1))
    println(solution.solution(dots2))
}